### PR TITLE
CSL-1247 put explorer into csl

### PR DESCRIPTION
**Pull request by ksaric**
https://issues.serokell.io/issue/CSL-1247

As is, merge from `explorer` `master`.
Used `git subtree add -P explorer https://github.com/input-output-hk/cardano-sl-explorer master`.

**EDIT** do not merge before @domenkozar gives us green light for Nix build feature (TODO add link)

### CODE REVIEW

**Comment by ksaric** [comment]
Not sure whether we should put the **explorer** Haskell source in the `cardano-node` _src_ directory or not?
Similar to the wallet?

**Comment by gromakovsky** [comment]
@ksaric 
> Not sure whether we should put the explorer Haskell source in the cardano-node src directory or not?

No, let's keep it in `explorer` folder. 

>  Similar to the wallet?

Sooner or later we will move wallet in `wallet` folder.

> Can I add the modules to cardano-node or will it get in the way there?

We should put as much explorer logic to `explorer` directory as possible.

File: explorer/LICENSE [hunk 1]
```diff
@@ -1,20 +0,0 @@
-Copyright (c) 2016 IOHK
```
Comment (gromakovsky) [inline explorer/LICENSE]: We have a `LICENSE` file in all directories with different packages (`tools`, `update`, etc.). I think it should be in `explorer` too.

**Comment by gromakovsky** [comment]
1. Something should be done with CI. AFAIU, currently it doesn't build explorer.
2. I think `scripts/build/cardano-sl.sh` should build explorer too if `--explorer` is passed.

**Comment by akegalj** [comment]
@ksaric 
> Something should be done with CI. AFAIU, currently it doesn't build explorer.
@domenkozar said he can help us with CI if help is needed

File: explorer/LICENSE (hunk 1 above)
Comment (ksaric) [inline explorer/LICENSE]: Ok, will return the `LICENCE` back.

**Comment by domenkozar** [comment]
Indeed. Before we add explorer, we should switch to building packages with Nix first. Hopefully this week.

**Pull request closed without merging**